### PR TITLE
feature/CVSB-469

### PR DESCRIPTION
**Pull request by danieliordache**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
     "cordova-sqlite-storage": "^2.6.0",
     "cordova-wheel-selector-plugin": "^1.1.1",
     "es6-promise-plugin": "^4.2.2",
+    "flat": "^4.1.0",
```
Comment (vladddragomir) [inline package.json]: please remove dependency

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,23 +1,48 @@
...
 
-  constructor(platform: Platform, statusBar: StatusBar, splashScreen: SplashScreen, private alertCtrl: AlertController, private syncService: SyncService, private authService: AuthService, private mobileAccessibility: MobileAccessibility, private renderer: Renderer2) {
+  constructor(platform: Platform, statusBar: StatusBar, splashScreen: SplashScreen, private alertCtrl: AlertController, private syncService: SyncService, private authService: AuthService, private mobileAccessibility: MobileAccessibility, private renderer: Renderer2, public storageService: StorageService, public visitService: VisitService) {
     platform.ready().then(() => {
       statusBar.overlaysWebView(true);
       statusBar.styleLightContent();
-      splashScreen.hide();
+
+
+      this.storageService.read('state').then(
```
Comment (vladddragomir) [inline src/app/app.component.ts]: please use enum for 'state'

File: src/app/app.component.ts [hunk 3]
```diff
@@ -1,23 +1,48 @@
...
+        (resp) => {
+          let stateResp = resp;
+          if (stateResp) {
+            this.storageService.read('visit').then(
```
Comment (vladddragomir) [inline src/app/app.component.ts]: same for 'visit'

File: src/models/visit/visit.model.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { TestModel } from "../tests/test.model";
+
+export class VisitModel {
```
Comment (vladddragomir) [inline src/models/visit/visit.model.ts]: interface instead of class

File: src/pages/testing/test-creation/test-create/test-create.ts [hunk 5]
```diff
@@ -132,4 +125,25 @@ export class TestCreatePage implements OnInit {
...
+    });
+  }
+
+  addATFIssue(): void {
+    this.navCtrl.push('ATFIssuePage');
+  }
+
+  reviewTest(): void {
+    console.log('Visit: ', this.visitService.visit);
+    // this.navCtrl.push('TestSummaryPage', {
```
Comment (vladddragomir) [inline src/pages/testing/test-creation/test-create/test-create.ts]: remove comments

File: src/pages/testing/vehicle/vehicle/vehicle-details/vehicle-details.ts [hunk 6]
```diff
@@ -12,38 +12,41 @@ import { DATE_FORMAT } from "../../../../../app/app.enums";
...
-          }
-        );
-    } else {
-      this.navCtrl.popTo(this.navCtrl.getByIndex(this.navCtrl.length() - 3));
-    }
+  ionViewWillEnter() {
+    this.viewCtrl.setBackButtonText('Identify Vehicle');
   }
 
+  // addVehicle(): void {
```
Comment (vladddragomir) [inline src/pages/testing/vehicle/vehicle/vehicle-details/vehicle-details.ts]: remove commented function

File: src/providers/global/state-reforming.service.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    this.stateHistory = [];
+    for(let i = 0; i < nav.length(); i++) {
+      const view = nav.getByIndex(i);
+      this.stateHistory.push({
+        page: view.name,
+        params: view.data
+      });
+    }
+    let stateJSON = JSON.stringify(this.stateHistory);
+    this.storageService.update('state', stateJSON);
```
Comment (vladddragomir) [inline src/providers/global/state-reforming.service.ts]: please use enum for 'state'

File: test-config/services-mocks/visit-service.mock.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    })
+  }
+
+  public getTests(): TestModel[] {
+    return this.visit.tests;
+  }
+
+  public updateVisit() {
+    return true
+    // this.storageService.update('visit', this.visit);
```
Comment (vladddragomir) [inline test-config/services-mocks/visit-service.mock.ts]: remove comment

File: package.json (hunk 1 above)
Comment (danieliordache) [reply package.json]: done

File: src/app/app.component.ts (hunk 3 above)
Comment (danieliordache) [reply src/app/app.component.ts]: done

File: src/pages/testing/test-creation/test-create/test-create.ts (hunk 5 above)
Comment (danieliordache) [reply src/pages/testing/test-creation/test-create/test-create.ts]: done

File: src/providers/global/state-reforming.service.ts (hunk 7 above)
Comment (danieliordache) [reply src/providers/global/state-reforming.service.ts]: done

File: test-config/services-mocks/visit-service.mock.ts (hunk 8 above)
Comment (danieliordache) [reply test-config/services-mocks/visit-service.mock.ts]: done

File: src/models/tests/test-type.model.ts [hunk 9]
```diff
@@ -8,15 +8,15 @@ export interface TestTypeModel {
   id?: string;
   certificateNumber?: string;
   lecCertificateNumber?: string;
-  expiryDate?: Date;
-  startTime: Date;
-  endTime?: Date;
+  expiryDate?: Date | string;
```
[the remark this answers is not in the document]
Comment (danieliordache) [reply src/models/tests/test-type.model.ts]: done

File: src/models/tests/test.model.ts [hunk 10]
```diff
@@ -2,11 +2,10 @@ import { PreparersModel } from "../reference-data-models/preparers.model";
 import { TEST_REPORT_STATUSES } from "../../app/app.enums";
 import { VehicleModel } from "../vehicle/vehicle.model";
 
-export interface TestReportModel {
-  startTime: Date;
-  endTime: Date;
-  testStatus: TEST_REPORT_STATUSES;
-  cancellationReason: string;
+export interface TestModel {
+  startTime: Date | string;
```
[the remark this answers is not in the document]
Comment (danieliordache) [reply src/models/tests/test.model.ts]: fixed

File: test-config/services-mocks/test-service.mock.ts [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+  endTestReport(test: TestModel) {
+    test.endTime = new Date().toISOString();
+  }
+
+  addVehicle(test: TestModel, vehicle: VehicleModel) {
+    test.vehicles.push(vehicle);
+  }
+
+  getTestReportTitle(testReport: TestModel): string {
```
Comment (tartino) [inline test-config/services-mocks/test-service.mock.ts]: Is is possible to use a better structure for this...including the one above

File: src/providers/test/test.service.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+  endTestReport(test: TestModel) {
+    test.endTime = new Date().toISOString();
+  }
+
+  addVehicle(test: TestModel, vehicle: VehicleModel) {
+    test.vehicles.push(vehicle);
+  }
+
+  getTestReportTitle(testReport: TestModel): string {
```
Comment (tartino) [inline src/providers/test/test.service.ts]: Can you use a better structure

**Pull request merged**